src/main/java/org/mule/extension/db/internal/domain/connection/DefaultDbConnection.java

```diff
@@ -159,4 +190,113 @@ public boolean isTransactionActive() {
   public boolean supportsContentStreaming() {
     return true;
   }
+
+  public PreparedStatement prepareStatement(String sql) throws SQLException {
+    return jdbcConnection.prepareStatement(sql);
+  }
+
+  @Override
+  public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
+    resolveLobs(typeName, elements, new CollectionTypeResolver(this));
+    return jdbcConnection.createArrayOf(typeName, elements);
+  }
+
+  @Override
+  public Struct createStruct(String typeName, Object[] attributes) throws SQLException {
+    resolveLobs(typeName, attributes, new StructTypeResolver(this));
+    return jdbcConnection.createStruct(typeName, attributes);
+  }
+
+  public DatabaseMetaData getMetaData() throws SQLException {
+    return jdbcConnection.getMetaData();
+  }
+
+  private ResultSet getAttributes(String typeName) throws SQLException {
+    return getMetaData().getAttributes(jdbcConnection.getCatalog(), null, typeName, null);
+  }
+
+  protected void resolveLobs(String typeName, Object[] attributes, TypeResolver typeResolver) throws SQLException{
+    try {
+      Map<Integer, ResolvedDbType> dataTypes = getLobFieldsDataTypeInfo(typeName);
+
+      for (Map.Entry entry : dataTypes.entrySet()) {
+        Integer key = (Integer) entry.getKey();
+        ResolvedDbType dataType = (ResolvedDbType) entry.getValue();
+
+        typeResolver.resolveLobIn(attributes, key, dataType);
+      }
+    } catch (SQLException e) {
+      logger.warn("Unable to resolve lobs: {}. Proceeding with original attributes.", e.getMessage());
```

Comment:
if (logger.isWarnEnabled())

Reply:
but logger.debug method calls this.logger.logIfEnabled(FQCN, Level.DEBUG, bla)

Reply:
the purpose of the if is not to prevent the message being logger, but to avoid the generation of the string